documentation/design/proposals/bridging_external.adoc

```diff
@@ -0,0 +1,80 @@
+== Overview
+
+A feature requested by several users is the ability to connect address spaces to external AMQP endpoints. NOTE: this proposal does not cover bridging between address spaces on the same cluster.
+
+The use cases covered by this proposal are:
+
+1. Forwarding messages from a local queue in a local address space to a destination on a remote AMQP endpoint
+1. Forwarding messages to a local queue in a local address space from a destination on a remote AMQP endpoint
+1. Sending messages to a remote AMQP endpoint via a local address space - by creating a connector and using prefixing
+1. Receiving messages from a remote AMQP endpoint via a local address space - by creating a connector and using prefixing
+
+== Design
+
+The address space specification would be expanded with 2 addition sub-objects, `connectors` and `forwarders`. A connector configures the infrastructure to connect to a remote AMQP endpoint and add address matching rules with the specified addresses. A forwarder specifies message routing rules to/from a local address to a remote address.  
+
+The `AddressSpace` resource is extended in the following manner:
+
+```
+apiVersion: enmasse.io/v1beta1
+kind: AddressSpace
+metadata:
+  name: local
+spec:
+  type: standard
+  plan: standard-small
+  connectors:
+  - name: remote1
+    hostname: example.com
+    port: 5672
+    tls: # If not specified, do not enable TLS. Use just 'tls: {}' to enable TLS without any cert configuration.
+      caCertSecret:
+        name: remote1-secret # Secret containing CA to be trusted for this connector. NB! Secret must be readable by the system:serviceaccount:enmasse-infra:address-space-controller service account.
+      clientCertSecret:
+        name: client-remot1-secret # Secret containing client cert to be used for this connector. NB! Secret must be readable by the system:serviceaccount:enmasse-infra:address-space-controller service account.
+    credentials:
+      username: foo # Allow specifying credentials in plain text
+      password: bar
+      secret: # Allow specifying credentials in the secret using the k
+        name:  mysecret
+        usernameKeyRef: username # Optional, default is 'username'
+        passwordKeyRef: password # Optional, default is 'password'
+    addresses: # A list of remote addresses accessible via this address space. The addresses will be prefixed with the connector name (remote1/foo*). Addresses can be suffixed by a 'match all' wildcard.
+    - name: foo*
+  forwarders:
+  - name: fwd1
+    connector: remote1 # Reference to the connector used for the forwarding
+    localAddress: myqueue # Address within local address space
+    remoteAddress: foo/queue1 
+    direction: in | out # Specifies if messages are forwarded from localAddress to remoteAddress or from remoteAddress to localAddress.
+```
+
```

Comment:
On making a separate CR for the `connector`: I am a bit reluctant to add another CR mainly because I think we already have too many. We would also be adding this to satisfy some persona we haven't yet clearly defined. However, maybe we can get both the simplicity and the required complexity if we allow the connectors array to either be a `spec` or a `ref` to a connector resource? I would then argue that we hold off adding the extra CR until a user ("customer") comes along with the use case.

On making a remote-address and forwarder address type: I'm not sure why these should be separate types, but instead something like:

```
kind: Address
spec:
  address: myqueue // Previously known as localAddress
  type: forwarded
  properties:
    direction: in
    remoteAddress: foo/queue1
```

Another question is how do we know which connector to validate this address against? Should a reference to the connector also be in the properties?

Reply:
Document is updated and resolve after.